### PR TITLE
Fix empty-string error in shopping_cart::get_product_id_list

### PR DESCRIPTION
**Pull request by drbyte**
Ref: https://www.zen-cart.com/showthread.php?217915-shopping_cart-get_product_id_list-function

### CODE REVIEW

File: includes/classes/shopping_cart.php [hunk 1]
```diff
@@ -611,19 +611,20 @@ function remove_all() {
   }
   /**
    * Method return a comma separated list of all products in the cart
+   * NOTE: Not used in core ZC, but some plugins make use of it as a helper function
    *
    * @return string
-   * @todo ICW - is this actually used anywhere?
```
Comment (ajeh) [inline includes/classes/shopping_cart.php]: Just a heads up: I actually have shipping modules that use this function so while it appears un-needed it is usefully for obtaining the string of products_id in the cart.

Comment (drbyte) [inline includes/classes/shopping_cart.php]: Agreed. That's why I removed this comment and added another one above.

Comment (drbyte) [inline includes/classes/shopping_cart.php]: Even better ... I'll update it to mention shipping modules specifically, for future clarity.

**Comment by zcwilt** [comment]
maybe use a guard test

```php
function get_product_id_list() {
    if (!is_array($this->contents)) {
       return '';
    }
    reset($this->contents);
    $product_id_list = array();
    while (list($products_id, ) = each($this->contents)) {
      $product_id_list[] = $products_id;
    }
    return implode(',', $product_id_list);
  }
```

**Comment by drbyte** [comment]
Good point.

**Comment by zcwilt** [comment]
:+1:

**Comment by ajeh** [comment]
:+1:

**Pull request merged**